File: examples/typescript/sample.ts
import { captureException, flush, init } from "../../sdks/typescript/src/index.ts";

const apiKey = process.env.BUGBARN_API_KEY ?? "bb_live_example";
const endpoint = process.env.BUGBARN_ENDPOINT ?? "http://127.0.0.1:8080/api/v1/events";
const mode = process.env.BUGBARN_SAMPLE_MODE ?? "manual";

init({
  apiKey,
  endpoint,
});

async function run(): Promise<void> {
  if (mode === "uncaught") {
    setTimeout(() => {
      throw new Error("BugBarn TypeScript sample uncaught exception");
    }, 0);
    return;
  }

  await captureException(new Error("BugBarn TypeScript sample manual error"), {
    attributes: {
      "service.name": "bugbarn-ts-sample",
      "runtime.name": "node",
    },
    tags: {
      sample: "typescript",
    },
    extra: {
      mode: "manual",
    },
  });
  await flush();
}

void run().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});

File: examples/python/sample.py
from __future__ import annotations

import os

from bugbarn.client import capture_exception, init, shutdown


def main() -> None:
    api_key = os.environ.get("BUGBARN_API_KEY", "bb_live_example")
    endpoint = os.environ.get("BUGBARN_ENDPOINT", "http://127.0.0.1:8080/api/v1/events")
    mode = os.environ.get("BUGBARN_SAMPLE_MODE", "manual")

    init(api_key=api_key, endpoint=endpoint, install_excepthook=True)

    if mode == "uncaught":
        raise RuntimeError("BugBarn Python sample uncaught exception")

    capture_exception(
        RuntimeError("BugBarn Python sample manual error"),
        attributes={
            "service.name": "bugbarn-python-sample",
            "runtime.name": "python",
        },
        tags={"sample": "python"},
        extra={"mode": "manual"},
    )
    shutdown(timeout=2.0)


if __name__ == "__main__":
    main()
